Updated deprecated design in @GenericGenerator for Hibernate 6

MarekB01 opened the pull request:
Related to: https://github.com/BroadleafCommerce/QA/issues/4999

property strategy was replaced with type:

before: strategy="org.broadleafcommerce.common.persistence.IdOverrideTableGenerator"

after: type= IdOverrideTableGenerator.class

MarekB01 committed:
BroadleafCommerce/QA-4999_Update_of_deprecated_design_Hibernate_6

property strategy was replaced with type:

before: strategy="org.broadleafcommerce.common.persistence.IdOverrideTableGenerator"

after: type= IdOverrideTableGenerator.class

riteshadhikari17 reviewed (changes requested):
Your IDE is changing the order of imports when it's optimizing them. Could you please change the layout in settings and revert them?

MarekB01 committed:
BroadleafCommerce/QA-4999_Update_of_deprecated_design_Hibernate_6

property strategy was replaced with type:

before: strategy="org.broadleafcommerce.common.persistence.IdOverrideTableGenerator"

after: type= IdOverrideTableGenerator.class

Refactored code

riteshadhikari17 committed:
Fix typo
